apps/studio/components/interfaces/Auth/AuthProvidersFormValidation.tsx

```
@@ -1244,6 +1244,53 @@ const EXTERNAL_PROVIDER_TWITTER = {
+    EXTERNAL_X_EMAIL_OPTIONAL: {
+      title: 'Allow users without an email',
+      description:
+        'Allows the user to successfully authenticate when the provider does not return an email address.',
+      type: 'boolean',
+    },
+  },
+  validationSchema: object().shape({
+    EXTERNAL_X_ENABLED: boolean().required(),
+    EXTERNAL_X_CLIENT_ID: string().when('EXTERNAL_X_ENABLED', {
+      is: true,
+      then: (schema) => schema.required('Client ID is required'),
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)